lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -1815,6 +1815,17 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :transform => DefaultSource.method(:convert_to_regexp_list),
           :description => 'Define transactions you want the agent to ignore, by specifying a list of patterns matching the URI you want to ignore. For more detail, see [the docs on ignoring specific transactions](/docs/agents/ruby-agent/api-guides/ignoring-specific-transactions/#config-ignoring).'
         },
+        # Serverless
+        :'serverless_mode.enabled' => {
+          :default => false,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :transform => proc { |bool| NewRelic::Agent::ServerlessHandler.env_var_set? || bool },
+          :description => 'If `true`, the agent will operate in a streamlined mode suitable for use with short-lived ' \
+                          'serverless functions. NOTE: Only AWS Lambda functions are supported currently and this ' \
+                          "option is not intended for use without New Relic's Ruby Lambda layer offering."
```

Comment:
Sometimes I see links included when there is mention of a New Relic product/service. Would [this page](https://docs.newrelic.com/docs/serverless-function-monitoring/aws-lambda-monitoring/get-started/monitoring-aws-lambda-serverless-monitoring/) be appropriate to link to a word in "New Relic's Lambda layer offering"?

Reply:
This is a good idea. The Ruby agent specific documentation doesn't exist yet to make use of but that high level one will likely eventually offer a click-through path to the Ruby goods.

Please submit an inline suggestion in the format you're thinking of and I'll commit it. Alternatively, feel free to directly commit to the feature branch.